src/transformers/modeling_attn_mask_utils.py

```diff
@@ -276,7 +276,7 @@ def _ignore_causal_mask_sdpa(
         elif sliding_window is None or key_value_length < sliding_window:
             if len(attention_mask.shape) == 4:
                 return False
-            elif (is_training or not is_tracing) and torch.all(attention_mask == 1):
+            elif not is_tracing and torch.all(attention_mask == 1):
```

Comment:
It's actually a paradox branch where `is_tracing` and `torch.all(attention_mask == 1)` can never exist together